routes/calendars.js

```diff
@@ -22,4 +22,75 @@ router.get("/:id", async (req, res, next) => {
   }
 });
 
-module.exports = router;
\ No newline at end of file
+router.get("/", async (req, res, next) => {
+  const calendar = await CalendarDAO.getAll();
+  if (calendar) {
+    res.json(calendar);
+  } else {
+    res.sendStatus(404);
+  }
+});
+
+router.put("/:id", async (req, res, next) => {
+  const { name } = req.body;
+  await CalendarDAO.put(req.params.id, name);
+  res.sendStatus(200);
+});
+
+router.delete("/:id", async (req, res, next) => {
+  await CalendarDAO.delete(req.params.id);
+  res.sendStatus(200);
+});
+
+// **************************
+// POST /calendars/:id/events - creates an event for the specified calendar using JSON from the request body
+router.post("/:id/events", async (req, res, next) => {
+  const { name, date } = req.body;
+  if (!name || !date) {
+    res.status(400).send('body parameter "name" and "date" are required"');
+  } else {
+    const event0 = await CalendarDAO.createEvent(req.params.id, name, date);
+    res.json(event0);
+  }
+});
+
+// GET /calendars/:id/events/:id - returns event with provided id from specified calendar
+router.get("/:calendar_id/events/:event_id", async (req, res, next) => {
```

Comment:
javascript variables are usually camelCase.
```suggestion
router.get("/:calendarId/events/:eventId", async (req, res, next) => {
```